lib/facter/kmod.rb

```diff
@@ -20,8 +20,10 @@
             Dir.foreach("/sys/module/#{directory}/parameters") do |param|
               next if ['.', '..'].include?(param)
 
-              kmod[directory]['parameters'][param] = File.read("/sys/module/#{directory}/parameters/#{param}").chomp
+              File.readable?("/sys/module/#{directory}/parameters/#{param}") && kmod[directory]['parameters'][param] = File.read("/sys/module/#{directory}/parameters/#{param}").chomp
```

Comment:
I think it might be more readable (pun intended) as:

```suggestion
              next unless File.readable?("/sys/module/#{directory}/parameters/#{param}")
              kmod[directory]['parameters'][param] = File.read("/sys/module/#{directory}/parameters/#{param}").chomp
```

Reply:
Ah, but that won't trigger the `EACCES` error you intend I guess.